src/plugins/installer.py

```diff
@@ -0,0 +1,92 @@
+import os
+import pathlib
+
+from helperFunctions.install import read_package_list_from_file, run_cmd_with_logging
+
+
+class AbstractPluginInstaller:
+    # Even if some functoins don't need self we want to have them nicely
+    # grouped in this class
+    # pylint:disable=no-self-use
+
+    skip_docker_env = os.getenv('FACT_INSTALLER_SKIP_DOCKER') is not None
+
+    def __init__(self, base_path: pathlib.Path, distribution, skip_docker=skip_docker_env):
+        self.base_path = base_path
+        self.distribution = distribution
+        self.build_path = base_path / 'build'
+        self.skip_docker = skip_docker
+
+    def install(self):
+        cwd = os.getcwd()
+        os.chdir(self.base_path)
+        self.install_system_packages()
+        self.install_pip_packages()
+        self.install_other_packages()
+
+        self.install_files()
+
+        self.build_path.mkdir(exist_ok=True)
+        os.chdir(self.build_path)
+        self.build()
+        run_cmd_with_logging(f'sudo rm -rf {self.build_path}')
+        os.chdir(self.base_path)
+
+        if not self.skip_docker:
+            self.install_docker_images()
+
+        self.do_last()
+
+        os.chdir(cwd)
+
+    def install_docker_images(self):
+        '''
+        Build/Pull docker images
+        '''
+
+    def install_system_packages(self):
+        '''
+        Install packages with apt/dnf
+        '''
+        build_pkg_path = pathlib.Path('./apt-pkgs-build.txt' if self.distribution != 'fedora' else './dnf-pkgs-build.txt')
+        runtime_pkg_path = pathlib.Path('./apt-pkgs-runtime.txt' if self.distribution != 'fedora' else './dnf-pkgs-runtime.txt')
+
+        pkg_list = read_package_list_from_file(build_pkg_path) if build_pkg_path.exists() else [] \
+            + read_package_list_from_file(runtime_pkg_path) if runtime_pkg_path.exists() else []
```

Comment:
I personally dislike having line breaks with "\\" in the code and try to avoid them where possible
making a function from the snippet could also reduce code duplication:
```suggestion
        pkg_list = self._read_packages(build_pkg_path) + self._read_packages(runtime_pkg_path)

(...)
                
def _read_packages(package_file: Path) -> List(str):
    return read_package_list_from_file(package_file) if build_pkg_path.exists() else []
```